mlflow/store/artifact/unity_catalog_oss_models_artifact_repo.py

```diff
@@ -0,0 +1,161 @@
+)
+from mlflow.protos.unity_catalog_oss_messages_pb2 import (
+    TemporaryCredentials,
+)
+from mlflow.protos.unity_catalog_oss_service_pb2 import UnityCatalogService
+from mlflow.store._unity_catalog.lineage.constants import _DATABRICKS_LINEAGE_ID_HEADER
+from mlflow.store.artifact.artifact_repo import ArtifactRepository
+from mlflow.store.artifact.utils.models import (
+    get_model_name_and_version,
+)
+from mlflow.utils._spark_utils import _get_active_spark_session
+from mlflow.utils._unity_catalog_utils import (
+    get_artifact_repo_from_storage_info,
+    get_full_name_from_sc,
+)
+from mlflow.utils.oss_utils import get_oss_host_creds
+from mlflow.utils.proto_json_utils import message_to_json
+from mlflow.utils.rest_utils import (
+    _REST_API_PATH_PREFIX,
+    _UC_OSS_REST_API_PATH_PREFIX,
+    call_endpoint,
+    extract_api_info_for_service,
+)
+from mlflow.utils.uri import (
+    _OSS_UNITY_CATALOG_SCHEME,
+    get_databricks_profile_uri_from_artifact_uri,
+    get_db_info_from_uri,
+    is_oss_unity_catalog_uri,
+)
+
+_METHOD_TO_INFO = extract_api_info_for_service(UcModelRegistryService, _REST_API_PATH_PREFIX)
+_METHOD_TO_INFO_OSS = extract_api_info_for_service(
+    UnityCatalogService, _UC_OSS_REST_API_PATH_PREFIX
+)
+
+
+class UnityCatalogOSSModelsArtifactRepository(ArtifactRepository):
+    """
+    Performs storage operations on artifacts controlled by a Unity Catalog model registry
+
+    Temporary scoped tokens for the appropriate cloud storage locations are fetched from the
+    remote backend and used to download model artifacts.
+
+    The artifact_uri is expected to be of the form `models:/<model_name>/<model_version>`
+
+    Note : This artifact repository is meant is to be instantiated by the ModelsArtifactRepository
+    when the client is pointing to a Unity Catalog model registry.
+    """
+
+    def __init__(self, artifact_uri, registry_uri):
+        if not is_oss_unity_catalog_uri(registry_uri):
+            raise MlflowException(
+                message="Attempted to instantiate an artifact repo to access models in the "
+                f"OSSS Unity Catalog with non-Unity Catalog registry URI '{registry_uri}'. "
+                f"Please specify a Unity Catalog registry URI of the "
+                f"form '{_OSS_UNITY_CATALOG_SCHEME}[://profile]', e.g. by calling "
+                f"mlflow.set_registry_uri('{_OSS_UNITY_CATALOG_SCHEME}') if using the "
+                f"MLflow Python client",
+                error_code=INVALID_PARAMETER_VALUE,
+            )
+        super().__init__(artifact_uri)
+        from mlflow.tracking.client import MlflowClient
+
+        registry_uri_from_artifact_uri = get_databricks_profile_uri_from_artifact_uri(
+            artifact_uri, result_scheme=_OSS_UNITY_CATALOG_SCHEME
+        )
+        if registry_uri_from_artifact_uri is not None:
+            registry_uri = registry_uri_from_artifact_uri
```

Comment:
```suggestion
        registry_uri = get_databricks_profile_uri_from_artifact_uri(
            artifact_uri, result_scheme=_OSS_UNITY_CATALOG_SCHEME
        ) or registry_uri
```
Can we combine these statements?